keras/models.py

```diff
@@ -22,6 +22,59 @@
 from .layers import containers
 
 
+def count_named_params(weights):
+    ctr = 0
+    for k, v in weights.items():
+        if isinstance(v, dict):
+            ctr += count_named_params(v)
+        elif isinstance(v, list):
+            ctr += len(v)
+        else:
+            raise ValueError('unrecognized weight structure of type'
+                             .format(type(v)))
+    return ctr
+
+
+def mk_hdf5_tree(root, weights):
+    '''Recursive function to save a weight dictionary to
+    an hdf5 file. To be used in conjuction with `save_named_weights`.
+    '''
+    for k, w in weights.items():
+        subroot = root.create_group(k)
+        if isinstance(w, dict):
+            mk_hdf5_tree(subroot, w)
+        else:
+            subroot.attrs['nb_params'] = len(w)
```

Comment:
The whole `nb_params` business is confusing. Are you aware of the `count_params` methods in models, which means something different? "parameter count" has a specific meaning with regard to neural networks: the total number of (scalar) weights, not the number of weight matrices.